File: src/main/kotlin/com/ks/culinario/network/dto/RecipeDTO.kt
package com.ks.culinario.network.dto

data class RecipeDTO(
    val id: Int,
    val title: String,
    val description: String,
    val step: RecipeStepDTO,
    val owner: String,
    val category: String,
    val created: String,
    val shoppingListId: Int?
)

data class RecipeStepDTO(
    val title: String,
    val description: String
)

File: dotnet/Culinario.Application/DTOs/RecipeDTO.cs
namespace Culinario.Application.DTOs;

public record RecipeDTO(
    int Id,
    string Title,
    string Description,
    RecipeStepDTO Step,
    string Owner,
    string Category,
    string Created,
    int ShoppingListId
);

public record NewRecipeDTO(
    string Title,
    string Description,
    RecipeStepDTO Step,
    string Owner,
    string Category,
    int ShoppingListId
);

public record RecipeStepDTO(string Title, string Description);
